### DISCUSSION TITLE
NACA0012 case diverging

### DISCUSSION THREAD

**Discussion by nipinl** [loganoz/horses3d#151] [Q&A] [answered]
First of all, I would like to thank you for the code Horses3D. I went through the paper and I would like to try it out for simulation of flow around aerofoil.

I ran the case NACA0012 available under Solver/test/NavierStokes directory and found that it is diverging after a few iterations(test case has number of iterations only 1 ). I tried to use FAS with BIRK5 etc but in vein. I would like to know if that is due to the mesh or I am missing something. I have attached the Control file here.

Any leads would be really helpful !!

Regards,
Nipin
[NACA0012.control.txt](https://github.com/loganoz/horses3d/files/11323789/NACA0012.control.txt)

**Comment by loganoz** [comment, marked as answer]
Hi @nipinl, 

Thanks for the input. As a general comment, the main objective of these test cases is to test the sanity of the code. That means checking that the residual after a small number of iterations is the same between commits. That means that most of the times we are not using the correct mesh/setup for a real simulation. For example, meshes tend to be smaller than required to get meaningful results. Also using Gauss-Lobatto nodes in the standard DGSEM is known to have some numerical issues. 

That is probably why you are getting wrong results (divergence) when increasing the number of iterations for this particular test case. Some things you could try to improve the robustness are: generate a better (more refined) mesh, use Gauss instead of Gauss-Lobatto // use a Split Form (check page 8 in the manual).

Please, keep me posted on how that goes. 
Best,
Gonzalo

**Reply by nipinl** [reply]
Hi @loganoz,

Thank you for the detailed reply. I too had the impression that the test cases are not to be seen as tutorials but I was not sure what make it less robust. Thank you for providing a clear direction on that. Yes, I'll keep you posed.

Cheers,
Nipin
